cmd/posix.go

```diff
@@ -896,7 +896,12 @@ func (s *posix) createFile(volume, path string) (f *os.File, err error) {
 		}
 	}
 
-	w, err := os.OpenFile(filePath, os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0666)
+	var w *os.File
+	if sync {
+		w, err = os.OpenFile(filePath, os.O_CREATE|os.O_WRONLY|os.O_SYNC, 0666)
```

Comment:
O_APPEND is missing here

Reply:
O_APPEND is not necessary it's a all write or no write operation.

Reply:
> O_APPEND is not necessary it's a all write or no write operation.

That's also what I thought but the old code contains O_APPEND, however you are right, this could be a bug (I remember seeing one user having duplicated json in format.json)

Can you then remove O_APPEND from else block ?

Reply:
O_APPEND is needed @vadmeste for writing part.1 files where we append erasure coded blocks to the end.

Reply:
We simply leveraged AppendFile previously, by not performing a second AppendFile operation.

Reply:
> We simply leveraged AppendFile previously, by not performing a second AppendFile operation.

Yeah I understand better now, can we rename createFile() to appendFile() here because that's what it confused me in the first place (createFile will assume that we are about to create a new file)

Reply:
It's a loaded function @vadmeste we can many variations like that, but it's not really that useful. Refactor like that I would like to avoid in this PR for the moment.

Reply:
To make it simply renamed createFile to openFile

Reply:
It is indeed confusing for the reviewers/maintainers to know why we are not doing O_APPEND when sync is enabled.

One thing that can be done is, take `flags` as argument instead of `sync`  similar to `os.OpenFile`

so AppendFile can call with flags = os.O_CREATE|os.O_APPEND|os.O_WRONLY
and WriteAll can call with flags = os.O_CREATE|os.O_WRONLY|os.O_SYNC